### PR TITLE
Increase wait time to avoid flaky test

### PR DESCRIPTION
**Pull request by Hailong-am**
### Description

Add more time to wait for master key been initialized
```
The ML encryption master key has not been initialized yet. Please retry after waiting for 10 seconds.
```
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/test/java/org/opensearch/integTest/ToolIntegrationTest.java [hunk 1]
```diff
@@ -47,7 +47,7 @@ public void setupTestAgent() throws IOException, InterruptedException {
             connectorId = setUpConnector();
         } catch (Exception e) {
             // Wait for ML encryption master key has been initialized
-            TimeUnit.SECONDS.sleep(10);
+            TimeUnit.SECONDS.sleep(15);
```
Comment (zane-neo) [inline src/test/java/org/opensearch/integTest/ToolIntegrationTest.java:50]: Can we make sure 15s is enough or shall we create a for loop query with sleep and max loop time until the initialization is completed?

Comment (Hailong-am) [reply src/test/java/org/opensearch/integTest/ToolIntegrationTest.java:50]: The `MLSyncUpCron` job run every 10s, increase to 15s  should good enough to wait for this job to complete and make master key initialized.

**Review by zane-neo — APPROVED** [review]

**Comment by zane-neo** [comment]
Test is failing: 
```
REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.integTest.VisualizationsToolIT.testVisualizationFound" -Dtests.seed=A7E9663F75410473 -Dtests.security.manager=false -Dtests.locale=sr-CS -Dtests.timezone=Europe/Belgrade -Druntime.java=11
```

**Comment by Hailong-am** [comment]
> Test is failing:
> 
> ```
> REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.integTest.VisualizationsToolIT.testVisualizationFound" -Dtests.seed=A7E9663F75410473 -Dtests.security.manager=false -Dtests.locale=sr-CS -Dtests.timezone=Europe/Belgrade -Druntime.java=11
> ```

due to NPE introduced by https://github.com/opensearch-project/ml-commons/pull/1911 waiting for @jngz-es to fix NPE.

File: src/test/java/org/opensearch/integTest/ToolIntegrationTest.java (hunk 1 above)
Comment (dbwiddis) [reply src/test/java/org/opensearch/integTest/ToolIntegrationTest.java:50]:
```
[2024-01-29T18:44:43,764][INFO ][o.o.n.Node               ] [88665a43e3fa] started
[2024-01-29T18:44:43,807][INFO ][o.o.g.GatewayService     ] [88665a43e3fa] recovered [0] indices into cluster_state
[2024-01-29T18:44:53,752][INFO ][o.o.m.a.MLModelAutoReDeployer] [88665a43e3fa] Index not found, not performing auto reloading!
[2024-01-29T18:44:53,753][INFO ][o.o.m.c.MLCommonsClusterManagerEventListener] [88665a43e3fa] Starting ML sync up job...
[2024-01-29T18:45:03,786][INFO ][o.o.p.PluginsService     ] [88665a43e3fa] PluginService:onIndexModule index:[.plugins-ml-config/sZK8J9DtQC6TiFxNFcfmDA]
[2024-01-29T18:45:03,879][INFO ][o.o.c.m.MetadataCreateIndexService] [88665a43e3fa] [.plugins-ml-config] creating index, cause [api], templates [], shards [1]/[1]
```

Comment (Hailong-am) [reply src/test/java/org/opensearch/integTest/ToolIntegrationTest.java:50]: hmm, let me changed to retry as @zane-neo suggested

**Comment by Hailong-am** [comment]
test case failed not related to this change
```
REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.integTest.NeuralSparseSearchToolIT.testNeuralSparseSearchToolInFlowAgent_withIllegalEmbeddingField_thenThrowException" -Dtests.seed=7C4C3F7C07267F08 -Dtests.security.manager=false -Dtests.locale=no-NO -Dtests.timezone=Asia/Gaza -Druntime.java=21

org.opensearch.integTest.NeuralSparseSearchToolIT > testNeuralSparseSearchToolInFlowAgent_withIllegalEmbeddingField_thenThrowException FAILED
    org.opensearch.client.ResponseException: method [DELETE], host [http://127.0.0.1:43481/], URI [/_plugins/_ml/models/uBgCaI0Bt5e8BGUB5kIr], status line [HTTP/1.1 500 Internal Server Error]
    {"error":{"root_cause":[{"type":"exception","reason":"Model cannot be deleted in deploying or deployed state. Try undeploy model first then delete"}],"type":"exception","reason":"Model cannot be deleted in deploying or deployed state. Try undeploy model first then delete"},"status":500}
        at __randomizedtesting.SeedInfo.seed([7C4C3F7C07267F08:D2577AD21C5A4FB4]:0)
        at app//org.opensearch.client.RestClient.convertResponse(RestClient.java:385)
        at app//org.opensearch.client.RestClient.performRequest(RestClient.java:355)
        at app//org.opensearch.client.RestClient.performRequest(RestClient.java:330)
        at app//org.opensearch.integTest.BaseAgentToolsIT.makeRequest(BaseAgentToolsIT.java:333)
        at app//org.opensearch.integTest.BaseAgentToolsIT.makeRequest(BaseAgentToolsIT.java:305)
        at app//org.opensearch.integTest.BaseAgentToolsIT.makeRequest(BaseAgentToolsIT.java:294)
        at app//org.opensearch.integTest.BaseAgentToolsIT.deleteModel(BaseAgentToolsIT.java:192)
        at app//org.opensearch.integTest.NeuralSparseSearchToolIT.tearDown(NeuralSparseSearchToolIT.java:91)
```

**Pull request merged**